lib/board.ex

```diff
@@ -0,0 +1,62 @@
+defmodule Board do
+
+  defstruct [:size, :cells, :marks]
+
+  def create_cells(size) do
+    Enum.to_list(1..size * size)
+  end
+
+  def place_mark(board, position, mark) do
+    %{board | cells: List.replace_at(board.cells, position - 1, mark)}
+  end
+
+  def rows(board) do
+    Enum.chunk_every(board.cells, board.size)
+  end
+
+  def columns(board) do
+    rows(board)
+    |> List.zip
+    |> Enum.map(&Tuple.to_list/1)
+  end
+
+  def diagonal_lines(board) do
+    [slice_diagonally(board.cells, board.size),
+     slice_diagonally(List.flatten(reverse_lines(rows(board))), board.size)]
```

Comment:
This seems intricate. I would pull out the entire `List.flatten(...)` as I might miss a crucial detail there